src/publisher/message-queues.ts

```diff
@@ -211,25 +243,26 @@ export class Queue extends MessageQueue {
    * @emits Queue#drain when all messages are sent.
    */
   async publishDrain(): Promise<void> {
-    await this._publishInternal(true);
+    await this._publishInternal(true, 'manual drain');
   }
 
   /**
    * Cancels any pending publishes and calls _publish immediately.
    *
    * Does _not_ attempt to further drain after one batch is sent.
    */
-  async publish(): Promise<void> {
-    await this._publishInternal(false);
+  async publish(reason: string): Promise<void> {
```

Comment:
since you're adding a param, wouldn't you need to add a default value to make it backwards-compatible or make it optional?

Reply:
Also, re: private methods, this library is pretty old and wasn't built to use them, so it would be pretty painful to add it now, imo. I'll definitely think about that going forward, but e.g. a lot of the unit tests assume that innards are accessible for mocking. I'd like to find a better way to do the unit tests too, but there are over 900 of them now 😅

Reply:
I went ahead and made those optional parameter changes.